vividus-facade-jira/src/main/java/org/vividus/jira/JiraFacade.java

```diff
@@ -38,49 +40,55 @@
     private static final String ISSUE = "issue/";
     private static final String ISSUE_ENDPOINT = REST_API_ENDPOINT + ISSUE;
 
-    private final JiraClient jiraClient;
+    private final JiraClientProvider jiraClientProvider;
 
-    public JiraFacade(JiraClient jiraClient)
+    public JiraFacade(JiraClientProvider jiraClientProvider)
     {
-        this.jiraClient = jiraClient;
+        this.jiraClientProvider = jiraClientProvider;
     }
 
-    public String createIssue(String issueBody) throws IOException
+    public String createIssue(String issueBody, Optional<String> jiraKey) throws IOException, JiraConfigurationException
```

Comment:
not clear what is jira key: user creates a new issue and API asks him to provide key which is generated dynamically
I guess it should be `jiraInstanceKey`